README.md

```diff
@@ -128,7 +128,7 @@ If this isn't sufficient for your design, please seek out experts to talk to.
   Nonces of zero-length are not compliant with the specification and *extremely insecure and must not be used.*
 * As AES-GCM uses an underlying counter mode you cannot encrypt more than 2<sup>36</sup>-32 bytes at once as it will cause the counter to overflow.
 * AAD is also restricted in length to 2<sup>61</sup>-1 bytes due to internal encodings of the data
-* If you use different length tags with the same key, you lower the security of *all* tags produced by that key, not just the short ones. (TODO: Add reference)
+* If you use different length tags with the same key, you lower the security of *all* tags produced by that key, not just the short ones. ([Authentication weaknesses in GCM](https://csrc.nist.gov/csrc/media/projects/block-cipher-techniques/documents/bcm/comments/cwc-gcm/ferguson2.pdf))
```

Comment:
Thank you very much! Just a tweak to better match the flow/style of the existing text.

```suggestion
* If you use different length tags with the same key, you lower the security of *all* tags produced by that key, not just the short ones. (See [Authentication weaknesses in GCM](https://csrc.nist.gov/csrc/media/projects/block-cipher-techniques/documents/bcm/comments/cwc-gcm/ferguson2.pdf) by Niels Ferguson for that and other interesting issues with the construction.)
```